client/lib/service-worker/service-worker.js

```diff
@@ -140,3 +123,16 @@ self.addEventListener( 'fetch', function( event ) {
 	}
 } );
 /* eslint-enable */
+
+function precache() {
+	// Load configuration from server
+	return fetch( '/assets.json' ).then( function( response ) {
```

Comment:
can we just bake this into the service worker instead of fetching it?

Reply:
IMO, as @Tug mentioned, it is better to err on the side of infrequent changes to the service worker and to load necessary state (e.g. latest asset URLs) from the server. This also lets us do fancy tricks like detecting if the assets on the server differ or match what we have in the SW cache, and use postMessage to prompt the user to reload Calypso (via a link or button) when the new assets are downloaded and ready to use.

Reply:
Sounds good. Our expectation (hope?) is that most of calypso doesn't change on every build. It sounds like the way to maximize the value of the SW cache then is to do it this way, loading it dynamically.